plugins/workload-management/src/test/java/org/opensearch/plugin/wlm/QueryGroupTestUtils.java

```diff
@@ -0,0 +1,125 @@
+    }
+
+    public static ClusterState clusterState() {
+        final Metadata metadata = Metadata.builder().queryGroups(new HashSet<>(queryGroupList())).build();
+        return ClusterState.builder(new ClusterName("_name")).metadata(metadata).build();
+    }
+
+    public static Settings settings() {
+        return Settings.builder().build();
+    }
+
+    public static ClusterSettings clusterSettings() {
+        return new ClusterSettings(settings(), ClusterSettings.BUILT_IN_CLUSTER_SETTINGS);
+    }
+
+    public static QueryGroupPersistenceService queryGroupPersistenceService() {
+        ClusterService clusterService = new ClusterService(settings(), clusterSettings(), mock(ThreadPool.class));
+        return new QueryGroupPersistenceService(clusterService, settings(), clusterSettings());
+    }
+
+    public static List<Object> prepareSandboxPersistenceService(List<QueryGroup> queryGroups) {
+        Metadata metadata = Metadata.builder().queryGroups(new HashSet<>(queryGroups)).build();
+        Settings settings = Settings.builder().build();
+        ClusterSettings clusterSettings = new ClusterSettings(settings, ClusterSettings.BUILT_IN_CLUSTER_SETTINGS);
+        ClusterService clusterService = new ClusterService(settings, clusterSettings, mock(ThreadPool.class));
+        ClusterState clusterState = ClusterState.builder(new ClusterName("_name")).metadata(metadata).build();
+        QueryGroupPersistenceService queryGroupPersistenceService = new QueryGroupPersistenceService(
+            clusterService,
+            settings,
+            clusterSettings
+        );
+        return List.of(queryGroupPersistenceService, clusterState);
+    }
+
+    public static void compareResourceLimits(Map<String, Object> resourceLimitMapOne, Map<String, Object> resourceLimitMapTwo) {
+        for (String resourceName : resourceLimitMapOne.keySet()) {
+            assertTrue(resourceLimitMapTwo.containsKey(resourceName));
+            assertEquals(resourceLimitMapOne.get(resourceName), resourceLimitMapTwo.get(resourceName));
+        }
+    }
+
+    public static void compareQueryGroups(List<QueryGroup> listOne, List<QueryGroup> listTwo) {
+        assertEquals(listOne.size(), listTwo.size());
+        for (QueryGroup groupOne : listOne) {
+            String groupOneName = groupOne.getName();
+            List<QueryGroup> groupTwoList = listTwo.stream().filter(sb -> sb.getName().equals(groupOneName)).collect(Collectors.toList());
+            assertEquals(1, groupTwoList.size());
+            QueryGroup groupTwo = groupTwoList.get(0);
+            assertEquals(groupOne.getName(), groupTwo.getName());
+            assertEquals(groupOne.get_id(), groupTwo.get_id());
+            compareResourceLimits(groupOne.getResourceLimits(), groupTwo.getResourceLimits());
+            assertEquals(groupOne.getMode(), groupTwo.getMode());
+            assertEquals(groupOne.getUpdatedAtInMillis(), groupTwo.getUpdatedAtInMillis());
+        }
```

Comment:
How about sorting these two lists and then doing the index wise comparison I think that would be more efficient ?
If the query group already implements `equalsTo` method then we should leverage that.